### PR TITLE
Oceanwater 795 shader cleanup

### PR DESCRIPTION
**Pull request by mogumbo**
## Linked Issues:
| Jira Ticket 🎟️ | [OCEANWATER-795](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-795) |

The PR shrinks code a bit by using arrays and fixes a few problems we didn't know we had. There were bugs uncovered by VIPER that just hadn't been aggravated in OceanWATERS yet, but they may have been eventually.

## Summary of Changes
* Cleaned up spotlight texture projection to keep texture oriented correctly relative to spotlight. Handle case where spotlight is disabled by Ogre for a particular renderable.
* Use arrays wherever possible in shader and material files.
* Test for far edge of low-res sun shadow map, and cast no shadow beyond that.

## Test
* Visual simulation should behave as it did before.

### CODE REVIEW

File: models/terminator/materials/scripts/ow_terrain.frag [hunk 1]
```diff
@@ -226,14 +225,12 @@ void lighting(vec3 wsDirToSun, vec3 wsDirToEye, vec3 wsNormal, vec4 wsDetailNorm
...
-            spotlightParams0.xyz, spotlightColor0.rgb * spotlightIntensityScale0,
-            spotlightTexCoord0, vsDirToEye, vsDetailNormal, specular_power,
-            diffuse, specular);
-  spotlight(vsSpotlightPos1.xyz - vsPos, -vsSpotlightDir1.xyz, spotlightAtten0,
-            spotlightParams0.xyz, spotlightColor0.rgb * spotlightIntensityScale1,
-            spotlightTexCoord1, vsDirToEye, vsDetailNormal, specular_power,
-            diffuse, specular);
+  for (int i=0; i<2; i++) {
+    spotlight(vsLightPos[i+1].xyz - vsPos, -vsLightDir[i+1].xyz, spotlightAtten0,
+              spotlightParams[1], spotlightColor0.rgb * spotlightIntensityScale[i],
```
Comment (Samahu) [inline models/terminator/materials/scripts/ow_terrain.frag]: Does this `spotlightParams[1]` need to be `spotlightParams[i or i+1]` instead?

File: models/terminator/materials/scripts/ow_terrain.material [hunk 2]
```diff
@@ -33,39 +31,37 @@ fragment_program ow_terrain_frag glsl
...
-    param_named spotlightIntensityScale1  float 1.0
-    param_named_auto vsSpotlightPos0      light_position_view_space 1
-    param_named_auto vsSpotlightPos1      light_position_view_space 2
-    param_named_auto vsSpotlightDir0      light_direction_view_space 1
-    param_named_auto vsSpotlightDir1      light_direction_view_space 2
+    param_named spotlightIntensityScale[0]     float 1.0
+    param_named spotlightIntensityScale[1]     float 1.0
+    param_named_auto vsLightPos                light_position_view_space_array 3
+    param_named_auto vsLightDir                light_direction_view_space_array 3
+    param_named_auto spotlightParams          spotlight_params_array 3
```
Comment (Samahu) [inline models/terminator/materials/scripts/ow_terrain.material]:
```suggestion
    param_named_auto spotlightParams           spotlight_params_array 3
```

**Review by Samahu — APPROVED** [review]

File: models/terminator/materials/scripts/ow_terrain.frag (hunk 1 above)
Comment (mogumbo) [reply models/terminator/materials/scripts/ow_terrain.frag]: Good catch! Fixed.

File: models/terminator/materials/scripts/ow_terrain.material (hunk 2 above)
Comment (mogumbo) [reply models/terminator/materials/scripts/ow_terrain.material]: fixed

**Review by AstroStucky — APPROVED** [review]
Visuals look good. Code looks good. Great work!

**Review by Samahu — APPROVED** [review]

**Review by AstroStucky — APPROVED** [review]

**Pull request merged**